### PR TITLE
Bugs fix 

### LINKED ISSUE #12: Incorrect hop floating window position when 'virtualedit` is empty
hop2

### LINKED ISSUE #13: Pressing <Esc> causes vim exit
To reproduce:

- start nvim
- `:help` 
- `:wincmd o` making help window the only window in nvim
- `:HopWord`
- Press <Esc> to cancel jumping
- The nvim exits, which is not expected 

hop3

### PR DESCRIPTION
**Pull request by voldikss**
Fix #12, fix #13

### CODE REVIEW

File: lua/hop/init.lua [hunk 1]
```diff
@@ -113,6 +116,7 @@ local function hint_with(mode, opts)
 
   -- keybindings
   keymap.create_jump_keymap(hint_buf_handle, opts)
+  vim.api.nvim_set_option('virtualedit', save_virtualedit)
```
Comment (phaazon) [inline lua/hop/init.lua:118]: This is nice, thanks a lot!

File: lua/hop/keymap.lua [hunk 2]
```diff
@@ -17,7 +17,7 @@ function M.create_jump_keymap(buf_handle, opts)
     vim.api.nvim_buf_set_keymap(buf_handle, '', key, "<cmd>lua require'hop'.refine_hints(0, '" .. key .. "')<cr>", { nowait = true })
   end
 
-  vim.api.nvim_buf_set_keymap(buf_handle, '', '<esc>', '<cmd>q<cr>', {})
+  vim.api.nvim_buf_set_keymap(buf_handle, '', '<esc>', '<cmd>close<cr>', {})
```
Comment (phaazon) [inline lua/hop/keymap.lua:20]: What’s the difference between `close` and `quit` here?

**Review by phaazon — APPROVED** [review]

File: lua/hop/keymap.lua (hunk 2 above)
Comment (voldikss) [reply lua/hop/keymap.lua:20]: As the document says

```
							*:q* *:quit*
:q[uit]			Quit the current window.  Quit Vim if this is the last
			window(my note: not including help window and preview window). 
                        This fails when changes have been made and
			Vim refuses to |abandon| the current buffer, and when
			the last file in the argument list has not been
			edited.
```

If the `help` window is the only window in Vim, then after using `HopWord`, the hop window will be the last window, in that case if using `:q` to quit hop window, vim will exit.

File: lua/hop/init.lua [hunk 3]
```diff
@@ -45,6 +45,8 @@ local function hint_with(mode, opts)
 
   local winblend = opts and opts.winblend or defaults.winblend
 
+  local save_virtualedit = vim.api.nvim_get_option('virtualedit')
+  vim.api.nvim_set_option('virtualedit', 'all')
```
Comment (phaazon) [inline lua/hop/init.lua:49]: I think this caused a regression, as now the buffer is in `virtualedit` mode even though it wasn’t.

Comment (voldikss) [reply lua/hop/init.lua:49]: But we restore this option after opening hop window 
```vim
vim.api.nvim_set_option('virtualedit', save_virtualedit)
```

Comment (phaazon) [reply lua/hop/init.lua:49]: I think this is not the right way to do it? I reproduce every time:

1. Open any buffer.
2. Call any Hop commands.
3. When Hop quits, the window is now in `virtualedit`.

Do you remember why we needed that option? I don’t recall the need. :confused:

Comment (voldikss) [reply lua/hop/init.lua:49]: https://github.com/phaazon/hop.nvim/issues/12

**Pull request merged**